packages/api/src/command/webhook/fitbit.ts

```diff
@@ -32,80 +32,93 @@ const log = Util.log(`Fitbit Webhook`);
 export const processData = async (data: FitbitWebhook): Promise<void> => {
   console.log("Starting to process a Fitbit webhook.");
 
+  // Group notifications by Fitbit user ID
+  const fitbitUserGroupedNotifications: {
+    [fitbitUserId: string]: { collectionType: FitbitCollectionTypes; date: string }[];
+  } = {};
+
+  data.forEach(d => {
+    if (!fitbitUserGroupedNotifications[d.ownerId]) {
+      fitbitUserGroupedNotifications[d.ownerId] = [];
+    }
+    fitbitUserGroupedNotifications[d.ownerId].push({
+      collectionType: d.collectionType,
+      date: d.date,
+    });
+  });
+
+  // collect and group data by ConnectedUser
   const connectedUsersAndData = await Promise.all(
-    data.map(async d => {
-      const { collectionType, date, ownerId: fitbitUserId } = d;
-      try {
-        const connectedUsers = await getConnectedUsersByTokenOrFail(
-          ProviderSource.fitbit,
-          fitbitUserId
-        );
+    Object.entries(fitbitUserGroupedNotifications).map(async ([fitbitUserId, notifications]) => {
+      const connectedUser = await getConnectedUserByTokenOrFail(
+        ProviderSource.fitbit,
+        fitbitUserId
```

Comment:
OOC, why are changing from `getConnectedUsersByTokenOrFail` (plural) to `getConnectedUserByTokenOrFail` (singular)?

Reply:
With the current implementation, it is not possible that more than one ConnectedUser is connected to the same Fitbit user. I remember that you wanted to change both Fitbit and Withings to user the plural version, but I thought we'll just deal with that within the scope of that issue. It just seems wrong to use plural, and then use `users[0]`.

Reply:
The main problem with `getConnectedUserByTokenOrFail` is that we won't know if there are more than one record matching the filter. Sure, the current implementation should prevent that, but:
1. The filter is based on pre and post wild card, so it's easier to match something else if we send a small enough token (it should be super rare, I know);
2. In the case we introduce a bug and end up with more than one ConnectedUser per token, it will be harder to identify the issue (these things happen).

(not to mention the fact I brought up earlier about how bad for performance it is to use pre+post wildcard on DB queries)

With that being said, we could at least change the implementation of `getConnectedUserByTokenOrFail` to use `findAll` instead of `findOne` and check if there are more than one item on the result - in which case we can capture an error and still return the first element to keep the code running.

Reply:
Also, if we're moving away from the plural version for Fitbit, should we also do that on `/src/providers/fitbit/` and remove that function?

Reply:
I revisited that function implementation on `/src/providers/fitbit/`, and the flow is such that if a new connected user connects to a previously-connected fitbit user, they will receive their token, and then the other user's token will get revoked. This makes it so we do actually need the plural version of the function. 

The problem with the use of a wild card will be addressed in the corresponding issue.